internal/controller/scenario/scenario_controller.go

```diff
@@ -75,15 +77,30 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	}
 
 	var application *applicationapiv1alpha1.Application
-	err = retry.OnError(retry.DefaultRetry, func(_ error) bool { return true }, func() error {
+	var CustomRetry = wait.Backoff{
+		Steps:    5,
+		Duration: 10 * time.Second, // Default was 10 milliseconds
+		Factor:   1.0,
+		Jitter:   0.1,
+	}
+	err = retry.OnError(CustomRetry, func(_ error) bool { return true }, func() error {
 		application, err = r.getApplicationFromScenario(ctx, scenario)
 		if err != nil {
-			logger.Info("Failed to get Application from the IntegrationTestScenario, try again", "error:", err)
+			logger.Info("Failed to get Application from the IntegrationTestScenario, trying again", "error:", err)
 		}
 		return err
 	})
 	if err != nil {
-		logger.Error(err, "Failed to get Application from the IntegrationTestScenario after retry", "application", scenario.Spec.Application)
+		errString := "Failed to get Application from the IntegrationTestScenario even after retrying"
+		logger.Error(err, errString, "application", scenario.Spec.Application)
+		patch := client.MergeFrom(scenario.DeepCopy())
+		helpers.SetScenarioIntegrationStatusAsInvalid(scenario, errString)
+		err := r.Client.Status().Patch(ctx, scenario, patch)
+		if err != nil {
+			logger.Error(err, "Failed to update Scenario as Invalid")
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
```

Comment:
Because we continue processing in case the Application is not found, I think we can remove [this block of code](https://github.com/konflux-ci/integration-service/blob/main/internal/controller/scenario/scenario_adapter.go#L72-L84) from the scenario_adapter.go file, wdyt?

Reply:
Yup, that makes sense to me.

Reply:
Thanks! Removed